src/shared/mkfs-util.c

```diff
@@ -101,12 +102,25 @@ int make_filesystem(
         assert(fstype);
         assert(label);
 
+        if (root && !streq(fstype, "squashfs"))
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                       "Populating with source tree is only supported for squashfs");
```

Comment:
why not move this check down into the next series of if blocks? There, we already check for swap, and you add a check for squashfs. So you might as well check it there, and then even return EINVAL if root is specified for swap (since it really never makes sense there), and EOPNOTSUPP in the final else branch if root is set.